src/Security/Authorization/Policy/src/AuthorizationMiddleware.cs

```diff
@@ -61,8 +61,10 @@ public async Task Invoke(HttpContext context)
                 return;
             }
 
-            // Note that the resource will be null if there is no matched endpoint
-            var authorizeResult = await policyEvaluator.AuthorizeAsync(policy, authenticateResult, context, resource: endpoint);
+            // Note that the endpoint will be null if there is no matched endpoint
+            var authZContext = new AuthorizationMiddlewareContext { Context = context, Endpoint = endpoint };
+
+            var authorizeResult = await policyEvaluator.AuthorizeAsync(policy, authenticateResult, context, resource: authZContext);
```

Comment:
Isn't this a breaking change? IIRC @pranavkm looked at changing this in 3.x and decided to document it instead.

Reply:
Yeah this is a breaking change, but something we think its worth doing, as the workaround wouldn't be too hard for them to fix either, since they can get access to the endpoint via the context

Reply:
So looking at AuthorizeFilter, which today already flows HttpContext, if we just flow HttpContext instead of a new MiddlewareContext, you could almost argue this is a breaking change for consistency, as both EndpointRouting and MVC would return to always passing HttpContext as the resource...

Reply:
@pranavkm @blowdart @Tratcher thoughts on this?  Should we just flow the httpcontext like we used to in mvc, or introduce a new authz middleware context for the resource?

Reply:
I'm not sure the new AuthorizationMiddlewareContext is getting you anything since the AuthorizeAsync resource parameter remains object. You're also not adding any interesting data to the AuthorizationMiddlewareContext.

Reply:
Right, the only benefit is we could add stuff to it in the future, but doesn't seem really worth much over just giving the HttpContext like before